pkg/provider/cloud/aks/provider.go

```diff
@@ -225,7 +237,7 @@ func ConvertStatus(provisioningState string, powerState armcontainerservice.Code
 	case provisioningState == string(resources.SucceededAKSState) && powerState == armcontainerservice.Code(resources.StoppedAKSState):
 		return apiv2.StoppedExternalClusterState
 	case provisioningState == string(resources.FailedAKSState):
-		return apiv2.ErrorExternalClusterState
+		return apiv2.ErrorExternalClusterState // TODO: think about changing this to "Warning" and add other variables to decide if actual error, because there are cases that Azure states "Failed" for Provision State but actually it's working/usable
```

Comment:
```suggestion
		case provisioningState == string(resources.StartingAKSState):
		return apiv2.StartingExternalClusterState
```
lets add `apiv2.StartingExternalClusterState` for both cluster and md

Reply:
This case is already being handled by line 231:
```
	case provisioningState == string(resources.StartingAKSState):
		return apiv2.ProvisioningExternalClusterState
```

Reply:
Anyway, I decided to add a similar one for MD